src/core/Akka.Tests/Loggers/SemanticLoggingSpecs.cs

```diff
@@ -0,0 +1,384 @@
+//-----------------------------------------------------------------------
+// <copyright file="SemanticLoggingSpecs.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2025 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2025 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Akka.Configuration;
+using Akka.Event;
+using Akka.TestKit;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Tests.Loggers
+{
+    public class SemanticLoggingSpecs : AkkaSpec
+    {
+        public SemanticLoggingSpecs() : base(ConfigurationFactory.ParseString(@"
+            akka {
+                loglevel = INFO
+                stdout-loglevel = INFO
+            }
+        "))
+        {
+        }
+        [Fact(DisplayName = "MessageTemplateParser should parse positional templates correctly")]
+        public void MessageTemplateParser_should_parse_positional_templates()
+        {
+            var template = "Value is {0} and status {1}";
+            var propertyNames = MessageTemplateParser.GetPropertyNames(template);
+
+            propertyNames.Should().NotBeNull();
+            propertyNames.Should().HaveCount(2);
+            propertyNames[0].Should().Be("0");
+            propertyNames[1].Should().Be("1");
+        }
+
+        [Fact(DisplayName = "MessageTemplateParser should parse named templates correctly")]
+        public void MessageTemplateParser_should_parse_named_templates()
+        {
+            var template = "User {UserId} logged in from {IpAddress}";
+            var propertyNames = MessageTemplateParser.GetPropertyNames(template);
+
+            propertyNames.Should().NotBeNull();
+            propertyNames.Should().HaveCount(2);
+            propertyNames[0].Should().Be("UserId");
+            propertyNames[1].Should().Be("IpAddress");
+        }
+
+        [Fact(DisplayName = "MessageTemplateParser should handle escaped braces")]
+        public void MessageTemplateParser_should_handle_escaped_braces()
+        {
+            var template = "Use {{braces}} for {Value}";
+            var propertyNames = MessageTemplateParser.GetPropertyNames(template);
+
+            propertyNames.Should().NotBeNull();
+            propertyNames.Should().HaveCount(1);
+            propertyNames[0].Should().Be("Value");
+        }
+
+        [Fact(DisplayName = "MessageTemplateParser should handle format specifiers")]
```

Comment:
We should add a Verify spec for the log output when we do format specification or alignment specification